docs/tutorial-aks-demo.md

```diff
@@ -0,0 +1,386 @@
+# Secure AKS Secrets with Key Vault
+
+## Introduction
+
+A Secret in Kubernetes is meant to save sensitive and secure data like passwords, certificates, and keys. These Secret objects aren't secret or secure! They're encoded using Base64 and saved into *etcd*. Also, by default, any pod can access to them. As a solution, we can encrypt data at rest as explained in [Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data). In this case, it's recommended to use a Key Management System (KMS) to securely save the encryption keys.
+
+KMS systems themselves have the feature to save the sensitive data like keys, secrets, files, and certificates. So, why not simply using it instead! Even more than just encryption, they offer powerful features like key rotation and expiration date.
+
+Now, to access KMS, a password is needed and should be saved in a Secret object in *etcd*! We returned back to the same problem. Fortunately, cloud providers have a solution. Instead of using a password to access KMS, the managed Kubernetes cluster will be granted access to KMS. In Azure, for example, AKS can connect to Key Vault or SQL Database… using an Identity. This connection could be done using the open-source [project Pod Identity](https://github.com/Azure/aad-pod-identity). It uses Azure AD to create an Identity and assign the roles and resources.
+
+Now that we have access to Key Vault, we can use its SDK or REST API in the application to retrieve the secrets. The SDK has support for .NET, Java, Python, JS, Ruby, PHP, etc. Or we can retrieve the secrets from a mounted volume. Historically, in Azure, this solution was implemented through [Kubernetes Key Vault Flex Volume](https://github.com/Azure/kubernetes-keyvault-flexvol). Now it's being deprecated. The new solution is [Azure Key Vault provider for Secret Store CSI driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure). Which is the Azure implementation of [Secrets Store CSI driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver).
```

Comment:
Can you clarify Kubernetes version dependency on CSI vs. Flex volume driver?

Reply:
I just mentioned Flex Volume is now deprecated. What should we add more ? Do you mean dependency on 1.16 ?

Reply:
```
Historically, in Azure, this solution was implemented through [Kubernetes Key Vault Flex Volume](https://github.com/Azure/kubernetes-keyvault-flexvol) which is now deprecated for Kubernetes version 1.16+. [Azure Key Vault provider for Secret Store CSI driver](https://github.com/Azure/secrets-store-csi-driver-provider-azure) which is the Azure provider implementation for [Secrets Store CSI driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver) is the recommended solution for 1.16+.
```